include/s3select_functions.h

```diff
@@ -1545,217 +1536,15 @@ struct _fn_like : public base_function
 
     if(constant_state == true)
     {
-      escape_expr_val = esc->eval();
-      like_expr_val = like_expr->eval();
-
-      if (like_expr_val.type != value::value_En_t::STRING)  {
-        throw base_s3select_exception("like expression must be string");
-      }
-
-      if (escape_expr_val.type != value::value_En_t::STRING)  {
-        throw base_s3select_exception("escape expression must be string");
-      }
+      param_validation(esc, like_expr);
 
       std::vector<char> like_as_regex = transform(like_expr_val.str(), *escape_expr_val.str());
       std::string like_as_regex_str(like_as_regex.begin(), like_as_regex.end());
-      compiled_regex = std::regex(like_as_regex_str);
+      compiled_regex.reset(new RE2(like_as_regex_str));
```

Comment:
please use `std::make_unique` (instead of new)

Reply:
Done